### ISSUE TITLE
Intermittent Error - The specified CGI application encountered an error and the server terminated the process. 

### ISSUE THREAD

**Issue by pritambaldota** [aspnet/Home#1950]
Hi All I am facing intermittent "**The specified CGI application encountered an error and the server terminated the process.**" after hosting it on Azure Web Sites. On Local it works fine.

I gone through this [post ](https://blogs.msdn.microsoft.com/waws/2016/12/11/asp-net-core-slow-perf-issues-on-azure-websites/)and issue [#1182](https://github.com/aspnet/KestrelHttpServer/issues/1182), which solved the problem for sometime. After upgrading to new version I am still seeing this issue.

The site is running  React App and don't have any server side code except reading some settings file from Public Blob.

Program.cs
```
public static void Main(string[] args)
        {
            var host = new WebHostBuilder()
                .UseKestrel()
                .UseContentRoot(Directory.GetCurrentDirectory())
                .UseIISIntegration()
                .UseStartup<Startup>()
                .Build();

            host.Run();
        
```

When I look at logs i can see following -
```
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:30098 on the IPv6 loopback interface.
System.AggregateException: One or more errors occurred. (Error -4089 EAFNOSUPPORT address family not supported) ---> Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4089 EAFNOSUPPORT address family not supported
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListenerPrimary.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.ListenerPrimary.<StartAsync>d__11.MoveNext()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.KestrelEngine.CreateServer(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.KestrelServer.Start[TContext](IHttpApplication`1 application)
---> (Inner Exception #0) Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4089 EAFNOSUPPORT address family not supported
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.tcp_bind(UvTcpHandle handle, SockAddr& addr, Int32 flags)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvTcpHandle.Bind(ServerAddress address)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListenerPrimary.CreateListenSocket()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Listener.<>c.<StartAsync>b__6_0(Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.ListenerPrimary.<StartAsync>d__11.MoveNext()<---
```

And after some info logs -

```
fail: Microsoft.AspNetCore.Server.Kestrel[0]
      TcpListenerPrimary.OnConnection
Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4047 EPIPE broken pipe
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.Check(Int32 statusCode)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.Libuv.write2(UvRequest req, UvStreamHandle handle, uv_buf_t* bufs, Int32 nbufs, UvStreamHandle sendHandle, uv_write_cb cb)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvWriteReq.Write2(UvStreamHandle handle, ArraySegment`1 bufs, UvStreamHandle sendHandle, Action`4 callback, Object state)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.ListenerPrimary.DispatchConnection(UvStreamHandle socket)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.TcpListenerPrimary.OnConnection(UvStreamHandle listenSocket, Int32 status)
```

After doing few refresh site starts working.  

Project.json -
```
{
  "dependencies": {
    "Microsoft.NETCore.App": {
      "version": "1.1.0",
      "type": "platform"
    },
    "Microsoft.AspNetCore.Razor.Tools": {
      "version": "1.0.0-preview2-final",
      "type": "build"
    },
    "Microsoft.AspNetCore.Diagnostics": "1.1.0",
    "Microsoft.AspNetCore.Mvc": "1.1.0",
    "Microsoft.AspNetCore.Routing": "1.1.0",
    "Microsoft.AspNetCore.Server.IISIntegration": "1.1.0",
    "Microsoft.AspNetCore.Server.Kestrel": "1.1.0",
    "Microsoft.AspNetCore.StaticFiles": "1.1.0",
    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.1.0",
    "Microsoft.Extensions.Configuration.Json": "1.1.0",
    "Microsoft.Extensions.Logging": "1.1.0",
    "Microsoft.Extensions.Logging.Console": "1.1.0",
    "Microsoft.Extensions.Logging.Debug": "1.1.0",
    "Microsoft.Extensions.Options.ConfigurationExtensions": "1.1.0",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.1.0"
  },

  "tools": {
    "Microsoft.AspNetCore.Razor.Tools": "1.0.0-preview2-final",
    "Microsoft.AspNetCore.Server.IISIntegration.Tools": "1.0.0-preview2-final"
  },

  "frameworks": {
    "netcoreapp1.1": {
      "imports": [
        "dotnet5.6",
        "portable-net45+win8"
      ]
    }
  },

  "buildOptions": {
    "emitEntryPoint": true,
    "preserveCompilationContext": true,
    "define": []
  },

  "runtimeOptions": {
    "configProperties": {
      "System.GC.Server": true
    }
  },

  "publishOptions": {
    "include": [
      "wwwroot",
      "Views",
      "Areas/**/Views",
      "appsettings.json",
      "web.config"
    ],
      "exclude": [
        "wwwroot/node_modules",
        "wwwroot/dist",
        "wwwroot/src",
        "wwwroot/localResources",
        "wwwroot/webpack.config.*.js",
        "wwwroot/package.json",
        "wwwroot/_references.js",
        "wwwroot/.babelrc",
        "wwwroot/index.html",
        "bower_components"
      ]
  },

  "scripts": {
    "postpublish": [ "dotnet publish-iis --publish-folder %publish:OutputPath% --framework %publish:FullTargetFramework%" ]
  }
}

```

Am I missing something?

**Comment by Tratcher** [comment]
@halter73 @cesarbs

**Comment by halter73** [comment]
@pritambaldota Judging by the stack trace for the EPIPE exception, it looks like you didn't update the Kestrel.dll to 1.1.0 which fixes this issue. Could you double check that the Web App you have published in Azure is really using the 1.1.0 Kestrel DLL?

One easy way to do this is to go to "https://{yoursite}.scm.azurewebsites.net" and navigate to the "Debug console > PowerShell" tab. In the powershell prompt, if you enter

```
(ls D:\home\site\wwwroot\Microsoft.AspNetCore.Server.Kestrel.dll).VersionInfo
```

What output do you get?

Setting the Kestrel ThreadCount to 1 as suggested in the [post](https://blogs.msdn.microsoft.com/waws/2016/12/11/asp-net-core-slow-perf-issues-on-azure-websites/) you link to should also fix the issue if you can't upgrade the dll.

**Comment by pritambaldota** [comment]
@halter73  I can see the output of this is 1.1.0

ProductVersion   FileVersion      FileName                                     
--------------   -----------      --------                                     
1.1.0-rtm-22752  1.1.0.21115      D:\home\site\wwwroot\Microsoft.AspNetCore....

I will use ThreadCount to 1 approach as of now or will clear all assemblies and try to add it again.

**Comment by halter73** [comment]
You're now the second person to report seeing this issue after upgrading from 1.0.0 to 1.1.0 on Azure Web Apps, but that EPIPE stack trace doesn't look like it could come from 1.1.0... I've heard that Azure Web Apps does some trickery where even when you upload new DLLs it will continue to run your old app from a shadow copy, though this should be super short lived.

Setting ThreadCount to 1 will definitely fix the issue and almost certainly won't hurt the perf of your app (since the thread count only applies to IO threads). But if you can keep the default thread count after clearing the assemblies and report back if you still see this issue, I would really appreciate it.

**Comment by pritambaldota** [comment]
@halter73 yeah I have observed this several times, when I publish the code to new azure website instead of existing, it used to work but not with the existing ones. So I have to delete the web app and re-create it several times. By any chance, do you know how to get rid of shadow copy from azure web app?

**Comment by edmccaffrey** [comment]
We are seeing this error on 1.1.1. No shadow copy issues, as this happened on a new App Service as well. Using the ThreadCount = 1 setting still generates the GCI error.

This is output of the above PowerShell command.
1.1.1            1.1.1.30217      D:\home\site\wwwroot\Microsoft.AspNetCore....

**Comment by cvsekhar** [comment]
I am seeing the same error on 1.1.1 When we are having a traffic it throws The specified CGI application encountered an error and the server terminated the process

**Comment by mvanhalen** [comment]
I have seen this issue each time I publish and there is an update to the App Service dll. Reloading the url 2 times normally fixes it. I´m using the Microsoft.AspNetCore.All 2.0.0-preview1-final With VS 2017 preview 2 which includes the new kestrel version 2.0.0 preview 1 it seems. I will try a newer version of Microsoft.AspNetCore.All  to see if that works better

**Comment by raRaRa** [comment]
I'm running into similar issue. The thread count for my app rapidly goes up above 500 without any obvious reason, and the CGI error shows up. My only chance is to restart the app and it'll start to run normally.

This is definitely related to slightly more load, but I wasn't expecting it to scale so badly :(

Running on netcoreapp1.1, AspNetCore version 1.1.2.

**Comment by ncsugrad94** [comment]
We are running into the same issue. The thread count for our app goes up around 700 and the cgi error happens. We can make it happen very easily with very little traffic to our app,
We changed our HttpClient to be a static instance, and on netcoreapp1.1, AspNetCore version 1.1.2,
NETStandard.Library version 1.6.1.
We tried thread count=1, but backed it out when it had no impact. We are supposed to go live very soon but are blocked by this issue.

**Comment by halter73** [comment]
@raRaRa Did you [collect a memory dump](https://blogs.msdn.microsoft.com/kaushal/2017/05/04/azure-app-service-manually-collect-memory-dumps/) when you saw the high thread count? Doing that, and examining it in VS using the [Parallel Stacks Window](https://msdn.microsoft.com/en-us/library/dd998398.aspx) should show you exactly why your app's threadcount is going up rapidly. Once you resolve that, it's very likely that the CGI error will stop.

**Comment by halter73** [comment]
@ncsugrad94 I recommend that you also collect and analyze a memory dump taken when you see threadpool starvation in the way I describe in my last comment. Once you fix the threadpool starvation, I expect you'll see the CGI errors stop.

Kestrel's "ThreadCount" is completely unrelated to the number of threadpool threads and whether or not you'll experience threadpool starvation, so it makes sense that setting it to 1 would have no effect.

**Comment by raRaRa** [comment]
@halter73 Thank you for the detailed response, much appreciated.

I unfortunately didn't collect a memory dump. The main problem is that I usually can't spend much time looking into the problem when it occurs, since it affects my users. I haven't been able to reproduce it locally (yet).
My case is very similar to what @ncsugrad94 described. I have a static HttpClient for each API that I'm calling. (Instagram, Twitter, Dropbox, Facebook and Google). I have a theory that if one of the APIs are responding slowly then the connections might queue up and take up all of the available threads. Would it be worth checking into setting a low timeout for the HttpClient?

Thanks.

**Comment by halter73** [comment]
@raRaRa Lowering the timeouts for your HttpClients could certianly help alleviate threadpool starvation if you have a lot of threads blocked waiting on HTTP responses, but you shouldn't have any threads blocked on requests if you are fully utilizing async/await. I recommend capturing a memory dump the next chance you get, so you can verify the problem stems from long-lasting requests.

Instead of just lowering the timeouts, you should ensure that you are using async/await wherever possible. Long lasting requests might not be that bad if you can remove an unnecessary call to `Task.Result` that all your threads are blocked on. The Parallel Stacks Window can help you track down such calls to `Task.Result` and the like assuming they exist.

You might want to lower timeouts anyway to reduce the number of TCP connections that need to be kept open by the app to service concurrent requests, but you shouldn't need to lower the timeouts just to avoid threadpool starvation.

**Comment by ncsugrad94** [comment]
We found a few places that were using Task.result and missing async await. We fixed those and disabled most of our httpClient calls. The issue does not happen when running in VS 2015 on my laptop, only after we deploy to azure. I can recreate the issue by deploying our webpage and having 1 user open a bunch of tabs of the page (maybe 20). This is with most of our services disabled. Once the issue occurs all requests to the website fail and we have to restart the app service.

The service blew up to 700 threads after processing 989 requests, and it looks like it happened very suddenly, because  in the stdout log, everything looks very happy until the last few lines:

info: Microsoft.AspNetCore.Server.Kestrel[14]
      Connection id "0HL5T6R4JLL8O" communication error
Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4077 ECONNRESET connection reset by peer
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://dashboard-dev-d2.followmyhealth.com/api/token  
info: Microsoft.AspNetCore.Authentication.Cookies.CookieAuthenticationMiddleware[3]
      HttpContext.User merged via AutomaticAuthentication from authenticationScheme: DashboardAuthentication.
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://fmh-dashboard-dev-d2.azurewebsites.net/  
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[2]
      Request finished in 36.5641ms 302 
info: Microsoft.AspNetCore.Hosting.Internal.WebHost[1]
      Request starting HTTP/1.1 GET http://dashboard-dev-d2.followmyhealth.com/Login?session=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiIsImtpZCI6IkRlZmF1bHQifQ.eyJ1bmlxdWVfbmFtZSI6ImM1Y2MwNDYxLTU3OWUtNGVmMi1iMjNmLWE2YjkwMTNmNWIzYyIsIkZNSEF1dGhlbnRpY2F0aW9uVG9rZW5UeXBlIjoiSldUIiwiRk1ITGFzdExvZ2luIjoiNjM2MzQxNjcwNjE5MzUzMjg3IiwiRk1IUm9sZSI6IjI0NTc1NiIsImlzcyI6ImZvbGxvd215aGVhbHRoLmNvbSIsImF1ZCI6ImZvbGxvd215aGVhbHRoLmNvbSIsImV4cCI6MTQ5ODU3MTUxNCwibmJmIjoxNDk4NTcwNjE0fQ.MMIuzaTlxOhtzBdL5Ue2z6UhekrI5faEDYKUGs8koHk  
info: Microsoft.AspNetCore.Server.Kestrel[14]
      Connection id "0HL5T6R4JLL8Q" communication error
Microsoft.AspNetCore.Server.Kestrel.Internal.Networking.UvException: Error -4077 ECONNRESET connection reset by peer

We use the kudo explorer to see that dotnet.exe has mostly 20 to 30 threads steady but then all of a sudden jumps up to 500-900 threads. I am pretty sure there is nothing on our side causing this.

**Comment by raRaRa** [comment]
@ncsugrad94 Looks very similar to my problem. Here's a screenshot I took of the threads when the issue occurred. Note that one of the two servers ran just fine while the other one just jumped to 700 threads and was stuck until I restarted it. It was like that for 4 hours:
![Image](http://i.imgur.com/AcDxFSj.png)
I find it kinda weird that the threads jumps so high so quickly. stdout didn't show anything of interest. Can't wait to get a memory dump when it happens next.

**Comment by raRaRa** [comment]
And here's the number of TCP4 connections during that same time:
![](http://i.imgur.com/AYwrtAG.png)

My guess is that the Azure app wasn't responding quickly enough for a small period of time, causing a huge queue of connections. Could my guess be correct? But if that's the case, what's causing the huge number of threads, and why does it stay stuck as if it's in a deadlock until I restart the web app?

If a single TCP4 connection counts as one thread, then I can see why it's exploding. Now I'm wondering if it might be the Cors or auth middleware that is missing some potential async operations.

Thanks.

**Comment by ncsugrad94** [comment]
For what it is worth, through using procmon, we discovered some blocking done by a transient instance we were using for a token factory that accessed the azure keystore. We changed to singleton and so far no CGI error. It was doing a non await call to  AzureKeyStore.CreateAsync while processing each request. The hard part for us was it had been this way for almost 6+ months and we never had an issue until now.

**Comment by raRaRa** [comment]
I managed to get memory dump when the issue occurred today. I had to take memory dump from `dotnet.exe` to see the ~600 threads. `w3wp.exe` didn't show much information.

The problem is likely in the Azure Storage Table library. At least I can't see any problem with my code.
For Table queries I use `ExecuteQuerySegmentedAsync`. Everything is as async (with await of course) as it can be.

@halter73 Can I DM you the dump? I'm not super familiar on how to process the dump further, but it seems like 99% of the threads are blocking on `TableOperationHttpResponseParser`.

Image of the parallel stack:
![Image](http://i.imgur.com/RSCMBhi.png)

**Comment by raRaRa** [comment]
Seems to be related to https://github.com/Azure/azure-storage-net/issues/457

So the problem is that Azure Storage isn't asynchronous because of its OData dependency 😞

**Comment by yellowsix** [comment]
I am seeing a similar issue to @pritambaldota and described by others in this issue: very suddenly, the number of threads being used on one of my servers will jump up to 400 or 500, which crashes the server and then it restarts and is fine for a while.

I'm running an asp.net core 2.0 web application on Azure App Services. I just upgraded from 1.1 to 2.0, but the issue happened before the upgrade as well. I do use the table storage libraries mentioned by @raRaRa, but all of my attempts to capture a memory dump that points the finger at table storage have failed. Instead, I get a memory dump that looks like this when viewed with the parallel stack tool:

![image](https://user-images.githubusercontent.com/14003271/33010025-c6e66666-cd8d-11e7-9a2d-182d0045bd64.png)

Circled in red are 395 threads with no stack information, they are almost all in ntdll.dll, and a few are coreclr.dll, all say they are stopped in external code. I don't know if this means I just wasn't quick enough with capturing it (the problem spikes the server and then crashes the dotnet process in about 2 minutes, so it's very hard to capture it while it is happening), or if something else is going on...

I'm at a loss as to how to track this problem down further... I'm admittedly not that well versed in navigating the various logging and diagnostic tools. Anyone run into a similar issue or have any suggestions as to how to proceed?

**Comment by yellowsix** [comment]
Just to follow up... I was never able to pin down the cause of our asp.net core 2.0 application suddenly hanging every 30-40 minutes when under normal load (around 50k requests per hour across two instances in azure app services). I tried capturing memory dumps, std out logs, application insights... could not narrow it down to a single cause.

In frustration, I switched from targeting the netcoreapp2.0 framework to targeting net461 and win-x64 runtime. That seems to have fixed the problem! I really have no idea why... there is only one library we have that doesn't explicitly support netstandard2.0 -- Math.Net Numerics. And we only use some of the simpler things, like linear regression and line fitting. I tested all of those before deploying, and I checked the std out logs -- every request that used those methods finished successfully in under a second, so I don't think that was the culprit.

Wish I knew how to narrow this problem down further... spent a good five days on it... but for now we're just going to assume that running a website targeting netcoreapp2.0 on azure app services isn't stable enough for production scenarios.

**Comment by niemyjski** [comment]
I'm also seeing this kind of behavior. Our app gets 145million requests a day. It was running fine for the past 3 days and then today this behavior.

**Comment by halter73** [comment]
@niemyjski "The specified CGI application encountered an error and the server terminated the process" generally means that your app took more than two minutes to produce a response. Without more details, it's hard to tell what's causing the app to take so long to the respond.

Some of the above commenters found their issue was related to thread pool starvation. I would take a dump of the application when it's behaving poorly to see if that's the case for you too.

**Comment by niemyjski** [comment]
Definitely feels like thread starvation at least on one of the nodes. The
thing is this app has been running great for a few days and existing code (
asp.net legacy) is currently running without an issue (same code outside of
the controllers).

Thanks
-Blake Niemyjski

**Comment by halter73** [comment]
As mentioned in the above thread, when converting from legacy ASP.NET to ASP.NET core, switching from a synchronous API to an asynchronous API only to use `Task.Wait()` or `Task.Result` can perform far worse than the legacy code it replaced due to thread pool starvation.

In both cases the calling thread is blocked, but a synchronous method can continue to execute post-IO on the blocked calling thread. If you call an async method only to follow up with a call to `Task.Wait()/Result`, the async method oftentimes needs to finish running on a thread leased from the threadpool before it can complete its task. The async method can't continue to execute on the blocked calling thread since that's stuck in a call to `Task.Wait()/Result`.

**Comment by raRaRa** [comment]
I'm now seeing this issue again, after having fixed my thread exhaustion problem with Azure Storage. It happens more rarely now and there's nothing obvious from the metrics that could cause this issue for me.

My only theory is that I have an upload image feature on my website, and perhaps some users are uploading a large file, could that cause any issue?

**Comment by MaLiN2223** [comment]
@raRaRa I recently encountered the same problem as you did above. I managed to greatly reduce produced threads by tracking down and removing all non awaited async operations.

**Comment by yellowsix** [comment]
As another follow-up that may help others, I was able to narrow down the cause of my issue a bit further: it was a memory-related issue. We use an in-memory cache on the servers that is semi-permanent and contained a lot of objects, and a few large objects. Every so often, the web server would hang up in unpredictable places in the code, sometimes manifesting as described in this issue: a bunch of threads would pile up and block.

I was able to get a few more memory dumps during the issue, and all of them were stopped in the garbage collector... hard for me to say if it was coincidence or if the server was actually stuck in a really long garbage collection. A few of the memory dumps were stuck during deserialization of some large collections from JSON. Difficult to say for sure, but my suspicion was that the deserialization process temporarily spiked the memory requirement of that server, causing weird things to happen. I'd like to know why it manifested as a deadlocked server rather than a regular out-of-memory exception like if you e.g. try to allocate a dictionary that is too large... but I wasn't able to narrow the problem down further.

I was able to resolve the issue by relieving memory pressure on my web servers: I reduced the size of the cache and number of objects in the cache, and made deserialization more memory efficient. Haven't had a problem since.

**Comment by chenzhitong** [comment]
Looks like the problem I encountered, when my site is running for one weeks, will be inaccessible, the browser has not responded, and then request timeout. The Web site is hosted in a virtual machine. But I can use localhost on the server is normal access. It doesn't work when I restart website and restart IIS. It doesn't start working until I restarted the server.
Note: My site is HTTPS, and when the site is inaccessible due to the above issues, I change the protocol to HTTP and it is accessible.
Our website is open source https://github.com/neo-project/neo.org/

**Comment by Eilon** [comment]
Hi all, I'm closing this issue because it appears that some solutions have been offered for the problems that have been mentioned. If anyone is still having issues, please log a new issue with details and we will take a look at those specific problems. Thanks!
